apps/src/studio/studio.js

```diff
@@ -2397,25 +2425,42 @@ Studio.clearDebugRects = function() {
   $(".debugRect").remove();
 };
 
-
-Studio.drawWallTile = function (svg, wallVal, row, col) {
+Studio.drawWallTile = function (svg, wallVal, row, col, largeTile) {
   var srcRow, srcCol;
   if (wallVal == SquareType.WALL) {
     // use a random coordinate
     // TODO (cpirich): these should probably be chosen once at level load time
     // and we should allow the level/skin to set specific row/col max values
     // to ensure that reasonable tiles are chosen at random
-    srcRow = Math.floor(Math.random() * constants.WallCoordMax);
+    srcRow = Math.floor(Math.random() * constants.WallRandomCoordMax);
     // Since [0,0] is not a valid wall tile, ensure that we avoid column zero
     // when row zero was chosen at random
     srcCol = srcRow ?
-                Math.floor(Math.random() * constants.WallCoordMax) :
-                1 + Math.floor(Math.random() * (constants.WallCoordMax - 1));
+                Math.floor(Math.random() * constants.WallRandomCoordMax) :
+                1 + Math.floor(Math.random() * (constants.WallRandomCoordMax - 1));
   } else {
     srcRow = (wallVal & constants.WallCoordRowMask) >> constants.WallCoordRowShift;
     srcCol = (wallVal & constants.WallCoordColMask) >> constants.WallCoordColShift;
   }
 
+  // We might end up scaling this piece a little.  In that case, it's likely
+  // we'll also add an offset to its X/Y to keep it centered.
+  var multiplySize = 1;
+  var addOffset = 0;
+
+  if (skin.enlargeWallTiles && 
+      srcRow >= skin.enlargeWallTiles.minRow &&
+      srcRow <= skin.enlargeWallTiles.maxRow &&
+      srcCol >= skin.enlargeWallTiles.minCol &&
+      srcCol <= skin.enlargeWallTiles.maxCol) {
+    multiplySize = 1.4;
```

Comment:
This makes sense if the assets are what they are and can't be changed. But if we can request changes to the assets, it seems like it would be much cleaner not to have the code try to fix up the assets with magic constants like `1.4` and `-0.2`. Perhaps we can make it a goal to get a new drop of the assets that allows us to remove some of this special case code?